app/controllers/addFinancialInstitution/IsThisAddressController.scala

```diff
@@ -74,11 +74,14 @@ class IsThisAddressController @Inject() (
               formWithErrors =>
                 Future.successful(BadRequest(view(formWithErrors, mode, getFinancialInstitutionName(request.userAnswers), address.head.toAddress))),
               value =>
-                for {
-                  updatedAnswers                    <- Future.fromTry(ua.set(IsThisAddressPage, value))
-                  updatedAnswersWithSelectedAddress <- Future.fromTry(updatedAnswers.set(SelectedAddressLookupPage, address.head))
-                  _                                 <- sessionRepository.set(updatedAnswersWithSelectedAddress)
-                } yield Redirect(navigator.nextPage(IsThisAddressPage, mode, updatedAnswersWithSelectedAddress))
+                (address.head.country, ua.get(IsThisAddressPage)) match {
```

Comment:
what is the purpose of adding this match pattern where we catching all patterns anyway. I don't see the functional difference between what was before and this, only to add redundant pattern matching?